src/reflect/scala/reflect/internal/TreeGen.scala

```diff
@@ -732,11 +732,16 @@ abstract class TreeGen {
   def mkPatDef(pat: Tree, rhs: Tree)(implicit fresh: FreshNameCreator): List[ValDef] =
     mkPatDef(Modifiers(0), pat, rhs)
 
+  private def cpAtBoundAttachment(from: Tree, to: ValDef): to.type =
+    if (from.hasAttachment[AtBoundIdentifierAttachment.type]) to.updateAttachment(AtBoundIdentifierAttachment) else to
+  private def cpPatVarDefAttachments(from: Tree, to: ValDef): to.type =
```

Comment:
I assume this is UNIX for `copy`? It seems strange that `cpatBoundAttachment` only adds the attachment if `from` has it but `cpPatVarDefAttachments` unconditionally adds `PatVarDefAttachment`.

Reply:
One propagates whether the syntax was `x @ _`, the other adds that this is a patvar. There is doc over in TypeDiag.

I would justify `cp` by persistence of `mk` for make and similar, which are not due to a subculture. I do resist long names, and these are long.

I think I will shave my beard now and maybe go get a hair cut.